internal/component/loki/source/kubernetes/kubetail/tailer.go

```diff
@@ -292,9 +310,29 @@ func (t *tailer) tail(ctx context.Context, handler loki.EntryHandler) error {
 	}
 }
 
+// isJobPod determines if a pod is owned by a Job or CronJob workload.
+// Job pods have different lifecycle semantics than regular pods - they're
+// expected to run to completion and terminate, but we still want to collect
+// their logs even after termination.
+func isJobPod(pod *corev1.Pod) bool {
+	for _, ownerRef := range pod.GetOwnerReferences() {
+		if ownerRef.Controller != nil && *ownerRef.Controller {
+			// Check if owned by Job or CronJob
+			if ownerRef.Kind == "Job" || ownerRef.Kind == "CronJob" {
+				return true
+			}
+		}
+	}
+	return false
+}
+
 // containerTerminated determines whether the container this tailer was
 // watching has terminated and won't restart. If containerTerminated returns
 // true, it means that no more logs will appear for the watched target.
+//
+// This function implements standard Kubernetes restart policy logic and should
+// be used for regular pods. Job pods should use shouldStopTailingJobContainer()
+// instead, which has special handling for job lifecycle.
 func (t *tailer) containerTerminated(ctx context.Context) (terminated bool, err error) {
```

Comment:
Would it make sense to rename this to shouldStopTailingContainer to make it consistent with the new function added?